core/query/src/main/java/org/eclipse/rdf4j/query/QueryResults.java

```diff
@@ -87,6 +88,24 @@ public static Model asModel(CloseableIteration<? extends Statement, ? extends RD
 		return model;
 	}
 
+	/**
+	 * Returns a list of values of a particular variable out of the QueryResult.
+	 *
+	 * @param result
+	 * @param var    variable for which list of values needs to be returned
+	 * @return a list of Values of var
+	 * @throws QueryEvaluationException
+	 */
+	public static List<Value> getAllValues(TupleQueryResult result, String var) throws QueryEvaluationException {
+		try (Stream<BindingSet> stream = result.stream()) {
+			if (result.getBindingNames().contains(var)) {
+				return stream.map(bs -> bs.getValue(var)).collect(Collectors.toList());
+			} else {
+				return new ArrayList<Value>();
```

Comment:
A second way to reduce the amount of code even further would be to use a ternary operator, but that's just a matter of taste (some people like this style, some don't because it can be harder to read). 
I.e.  
```
return result.getBindingNames().contains(var) 
          ? stream.map(..
          : new ArrayList<>();
```

Reply:
Thank you @barthanssens . I will make these changes.

Reply:
Collections.emptyList() is a great option if we need an empty list.